libs/service-portal/access-control/delegations/src/hooks/useDomains.ts

```diff
@@ -80,25 +80,50 @@ export const useDomains = (includeDefaultOption = true) => {
     sessionStore.setItem('domain', name)
 
     const query = new URLSearchParams(location.search)
+    const currentDomain = query.get('domain')
 
-    if (query.get('domain')) {
+    if (currentDomain && currentDomain !== displayNameQueryParam) {
       query.set('domain', name)
       history.push(`${location.pathname}?${query.toString()}`)
```

Comment:
~IMO should use `history.replace` to not fill the navigation stack with entries when changing a dropdown.